src/main/java/com/hierynomus/smbj/share/DiskShare.java

```diff
@@ -466,6 +468,127 @@ private boolean exists(String path, EnumSet<SMB2CreateOptions> createOptions) th
         }
     }
 
+    /***
+     * Rename(Move) Directory/File. Default will using the false for replaceIfExist option
+     * @param path the path to the original directory/file
+     * @param newFileName the new directory/file name/path after rename
+     * @throws TransportException
+     * @throws SMBApiException
+     */
+    public void rename(String path, String newFileName) throws TransportException, SMBApiException {
```

Comment:
Ahh you misunderstood me. We have both a `File` and a `Directory` class which are returned from the `getFile` call in `DiskShare`. Why did you opt for putting the `rename` operation on the `DiskShare`, instead of implementing it in the `File` and `Directory` classes :)